tests: rpk redpanda start tests

r-vasquez opened the pull request:
## Cover letter

This PR introduces 4 small checks for `rpk redpanda start`:

1. Someone installs redpanda and starts it the first time.
2. Someone follows our guide, sets production mode, and starts redpanda.
3. Our new feature: `--mode dev-container`.
4. Passing a seastar flag

Fixes #5519

## Backport Required

- [x] not a bug fix

## UX changes
* none

## Release notes
* none

graphcareful commented on tests/rptest/services/redpanda.py:
```diff
@@ -992,6 +992,60 @@ def start_rp():
+            env_preamble = " ".join(
+                [f"{k}={v}" for (k, v) in self._environment.items()])
+
+            cmd = (f"{env_preamble} {self.find_binary('rpk')} redpanda start"
```
we can probably place this entire method nicely in the `rpk` class in `rptest/clients`

graphcareful commented on tests/rptest/services/redpanda.py:
```diff
@@ -992,6 +992,60 @@ def start_rp():
+                f"Redpanda service {node.account.hostname} failed to start within {timeout} sec using rpk",
+                retry_on_exc=True)
+
+        self.logger.debug(f"Node status prior to redpanda startup:")
```
no node status is printed here

r-vasquez replied:
The next line will call `start_service`, which calls `_log_node_process_state` at the beginning.

graphcareful commented on tests/rptest/services/redpanda.py:
```diff
@@ -992,6 +992,60 @@ def start_rp():
         self.start_service(node, start_rp)
         self._started.append(node)
 
+    def start_node_with_rpk(self, node, additional_args=""):
```
I see this method is very similar to `start_node`. I've vote for breaking out similar components into a private method, to reduce code duplication

r-vasquez replied:
Got it! moved rpk method to its own client and made is_status_ready a private method.

graphcareful commented on tests/rptest/tests/rpk_start_test.py:
```diff
@@ -0,0 +1,103 @@
+        pass
+
+    @cluster(num_nodes=1)
+    def test_simple_start(self):
```
awesome short but sweet test !

graphcareful commented on tests/rptest/tests/rpk_start_test.py:
```diff
@@ -0,0 +1,103 @@
+
+        # We execute checks when starting redpanda if production mode is enabled
+        assert self.redpanda.search_log("System check - PASSED")
+        self.redpanda.get_node_cpu_count
```
is there supposed to be an assertion here?

r-vasquez replied:
Ups, my fault.

r-vasquez commented:
[Force Push](https://github.com/redpanda-data/redpanda/compare/b2c4950ed982be0049c8554436a2cab405bb2703..6e5e3531bbe96f8bc1f9198ff4a0b9298e678153):

- Change `redpanda.start_node_with_rpk` to use `client/rpk_remote`.
- Abstract `is_status_ready` to a private function.

r-vasquez commented:
[Force Push](https://github.com/redpanda-data/redpanda/compare/6e5e3531bbe96f8bc1f9198ff4a0b9298e678153..d4be8bd9ca295568682e7786e17312846e64b0f4): Rebased and fixed conflicts, couldn't keep-base. No changes to the original code.

graphcareful reviewed (dismissed):
LGTM nice work !

r-vasquez commented:
Failure is: https://github.com/redpanda-data/redpanda/issues/5980

r-vasquez commented:
Failure in https://buildkite.com/redpanda/redpanda/builds/14604#0182cd10-c423-49da-8fc6-ddb0199c6c7f

Is a PR Blocker: https://github.com/redpanda-data/redpanda/issues/6147

jcsp commented on tests/rptest/clients/rpk_remote.py:
```diff
@@ -53,6 +53,13 @@ def tune(self, tuner):
+        return self._execute([
+            env_vars,
+            self._rpk_binary(), "redpanda", "start", "-v", additional_args,
+            ">>", log_file, "2>&1", "&"
```
Does ">> 2&1" (i.e. the `>>` but no filename) actually work if the default "" is set for log_file?

r-vasquez replied:
You are right, I'll make log_file required.

r-vasquez commented:
[Force Push](https://github.com/redpanda-data/redpanda/compare/cc82aa0f618d4da28a5dea5926663a40aee87e1f..2d694161f135af72c67d06d053d6e411d05d5b35): Make log_file required for `rpk_redpanda_start`
